crates/typst-library/src/compute/foundations.rs

```diff
@@ -83,6 +86,56 @@ pub fn panic(
     Err(msg)
 }
 
+
+/// Adds a warning to the output.
+///
+/// Message and hint are shown to the user (not rendered in the document).
+///
+/// Code executed after this statement is still run.
+/// If you wish to stop the compilation instead, see [`panic`]($panic).
+///
+/// # Example
+/// The code below produces a warning without aborting the compilation.
+/// ```typ
+/// #warn("something is not as expected")
+/// ```
+#[func]
+pub fn warn(
+    /// The virtual machine.
+    vm: &mut Vm,
+    /// The warning message.
+    message: Spanned<EcoString>,
+    /// An optional hint to display for the warning.
+    #[named]
+    hint: Option<EcoString>,
+) {
+    let Spanned { v: message, span } = message;
+
+    vm.vt.tracer.warn(SourceDiagnostic::warning(span, message)
+        .with_hints(hint.into_iter())
+        .with_emitter(span.id()));
+}
+
+/// Suppresses script-emitted warnings from the given module.
+///
+/// This can be useful if a package emits warnings that you do not care about.
+///
+/// # Example
+/// The code below suppresses warnings from the (purely fictional) package `"@preview/warnings:0.1.0"`.
+/// ```typ
+/// #import "@preview/warnings:0.1.0" as pick-your-name
+/// #nowarn(pick-your-name)
+/// ```
+#[func]
+pub fn nowarn(
```

Comment:
\@reviewer this name could be misleading, as it only suppresses warnings manually emitted in script, not warnings emitted by typst itself. I had no better idea however 😄